src/hotspot/share/classfile/classListParser.cpp

```diff
@@ -467,6 +467,10 @@ void ClassListParser::resolve_indy(Symbol* class_name_symbol, TRAPS) {
   Klass* klass = SystemDictionary::resolve_or_fail(class_name_symbol, class_loader, protection_domain, true, THREAD); // FIXME should really be just a lookup
```

Comment:
If an exception is unexpected, this should be CHECK not THREAD.

Reply:
I'll try CHECK and do more testing before posting an updated webrev.

Thanks for your review.

Reply:
For this change, don't change it to CHECK in case something new will fail, and any new exceptions will caught by checking k != NULL.  It just looks strange but can be cleaned up later.